lib/kitchen/transport/base.rb

```diff
@@ -138,9 +138,20 @@ def execute_with_retry(command, retryable_exit_codes = [], max_retries = 1, wait
         end
 
         def retry?(current_try, max_retries, retryable_exit_codes, exit_code)
-          current_try <= max_retries &&
-            !retryable_exit_codes.nil? &&
-            retryable_exit_codes.include?(exit_code)
+          retryable_exit_codes_exists = false
```

Comment:
What you are probably looking for is `Array(retryable_exit_codes).flatten.map(&:to_i).include?(exit_code)`